src/FTPClient.jl

```diff
@@ -16,6 +16,7 @@ type FTPClientError <: Exception
        lib_curl_error::UInt32
 end
 
+@enum FTP_MODE ascii_mode binary_mode
```

Comment:
@iamed2 I like that approach, but I like my functions have `mode::FTP_MODE` instead of `mode::Int`.
Would this work as the best of both worlds? Is there any obvious issues?
```
julia> module FTP_MODE
           @enum TRANSFER binary ascii
       end
FTP_MODE

julia> FTP_MODE.
TRANSFER ascii     binary    eval
julia> FTP_MODE.ascii
ascii

julia> typeof(FTP_MODE.ascii)
Enum FTP_MODE.TRANSFER:
binary = 0
ascii = 1
```
Now I can say `mode::FTP_MODE.TRANSFER`.